### ISSUE TITLE
userInfo does not work with login with email password

### ISSUE THREAD

**Issue by hdimessi** [auth0/Auth0.swift#728]
### Checklist

- [X] The issue can be reproduced in the [Auth0.swift sample app](https://github.com/auth0-samples/auth0-ios-swift-sample/tree/master/Sample-01) (or N/A).
- [X] I have looked into the [README](https://github.com/auth0/Auth0.swift#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/Auth0.swift/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/Auth0.swift/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/tags/c/sdks/5/swift) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

trying to get Auth0 userid for a logged in user so I m trying to call userInfo but it just won't work

```
Auth0
                .authentication(clientId: self.config.clientId, domain: self.config.domain)
                .login(usernameOrEmail: email,
                       password: password,
                       realm: "Username-Password-Authentication",
                       audience: "private.url")
                .start {
result in
                    switch result {
                    case .success(let credentials):
                        guard let token = credentials.accessToken,
                              let expireDate = credentials.expiresIn,
                              let tokenType = credentials.tokenType else {
                            let error = NSError(domain: "invalid token", code: 0)
                            promise(.failure(error))
                            return
                        }
                        let session = Session(token: token,
                                              expiresIn: expireDate,
                                              tokenType: tokenType)
                        let userSession = UserSession(session: session)
                        Auth0
                        .authentication(
                            clientId: self.config.clientId,
                            domain: self.config.domain
                        )
                        .userInfo(withAccessToken: token)
                        .start {
                            switch $0 {
                             case .failure(let error):
                                Log.error("FAILED to retrieve profile") // always fails here ALWAYS
                                print(error)
                             case .success(let profile):
                                let email = profile.email ?? "[unknown email]"
                                Log.debug("loaded profile for %@", email)

                                let identity = profile.sub

                                let _ = Cloud.shared.registerUser(
                                    identity,
                                    device: System.deviceId(),
                                    withAccessToken: token,
                                    withIdToken: credentials.idToken
                                )
                            }
                        }
                    case .failure(let error):
                        print("Failed with: \(error)")
                        promise(.failure(error))
                    }
                }
```

### Reproduction

basically copy/paste the code in the description

### Auth0.swift version

1.38.0

### Platform

macOS

### Platform version(s)

12.4

### Xcode version

13.4.1

### Package manager

SPM

**Comment by Widcket** [comment]
Hi @hdimessi, thanks for raising this.
What is the error message you're getting?

**Comment by hdimessi** [comment]
I forgot but it was something generic, nothing specific, I had to update to the latest version to get it to work actually
